packages/sw-routing/src/lib/navigation-route.js

```diff
@@ -69,9 +70,32 @@ class NavigationRoute extends Route {
     }
 
     const match = ({event, url}) => {
-      return event.request.mode === 'navigate' &&
-        whitelist.some((regExp) => regExp.test(url.pathname)) &&
-        !blacklist.some((regExp) => regExp.test(url.pathname));
+      let matched = false;
+      let message;
+
+      if (event.request.mode === 'navigate') {
+        if (whitelist.some((regExp) => regExp.test(url.pathname))) {
+          if (blacklist.some((regExp) => regExp.test(url.pathname))) {
+            message = `The navigation route is not being used, since while ` +
```

Comment:
Could we shorten this message?

`The navigation request is in both the whitelist and blacklist of the NavigationRoute.`